plugins/admission/pkg/opa/opa.go

```diff
@@ -22,46 +26,58 @@ func ProcessOPA(ctx context.Context, obj map[string]interface{}, resourceName, a
 		Version: opaVersion,
 	}
 	actionItems := make([]opa.ActionItem, 0)
-	for _, instanceObject := range configuration.OPA.CustomCheckInstances {
-		instance := opa.CustomCheckInstance{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: instanceObject.AdditionalData.Name,
-			},
-			Spec: opa.CustomCheckInstanceSpec{
-				CustomCheckName: instanceObject.CheckName,
-				Output:          instanceObject.AdditionalData.Output,
-				Parameters:      instanceObject.AdditionalData.Parameters,
-				Targets: funk.Map(instanceObject.Targets, func(s string) opa.KubeTarget {
-					splitValues := strings.Split(s, "/")
-					return opa.KubeTarget{
-						APIGroups: []string{splitValues[0]},
-						Kinds:     []string{splitValues[1]},
+	var allErrs error = nil
+	cluster := os.Getenv("FAIRWINDS_CLUSTER")
+	for _, check := range configuration.OPA.CustomChecks {
+		logrus.Debugf("Check %s is version %.1f\n", check.Name, check.Version)
+		switch check.Version {
+		case 1.0:
+			for _, instanceObject := range configuration.OPA.CustomCheckInstances {
```

Comment:
there's a lot of nesting inside this `switch` - I would suggest breaking each case out into its own function.

Reply:
@rbren I've broken out those `cases into `ProcessOPAV1` and `ProcessOPAV2` functions, would you mind taking a look?